esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1411,6 +1411,12 @@ void CEspHttpServer::sendVerifyCookieResponse(EspAuthRequest& authReq, CIArrayOf
     sendMessage(resp.str(), (format == ESPSerializationJSON) ? "application/json" : "text/xml");
 }
 
+bool CEspHttpServer::isValidUserName(const char *userName)
```

Comment:
The name suggests (to me) more than is intended. The implication of *valid* is closer to "this is a known name" than "this doesn't contain any characters precluding it from being a user name." Although not exactly the same, think about how XML distinguishes between valid (i.e., schema compliance) and well-formed (i.e., a parser won't choke on it).

As I read it top to bottom I was surprised that the empty string check wasn't included. I did see where the distinction between the two cases is relevant. I'm just emphasizing how *isValid* can be confusing.

Reply:
Change the isValid to isMalformed.